components/custom_board/tas5805m/tas5805m.c

```diff
@@ -0,0 +1,277 @@
+esp_err_t tas5805m_read_byte(uint8_t register_name, uint8_t *data)
+{
+
+  int ret;
+  i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+  i2c_master_start(cmd);
+  i2c_master_write_byte(cmd, TAS5805M_ADDRESS << 1 | WRITE_BIT, ACK_CHECK_EN);
+  i2c_master_write_byte(cmd, register_name, ACK_CHECK_EN);
+  i2c_master_stop(cmd);
+  ret = i2c_master_cmd_begin(I2C_TAS5805M_MASTER_NUM, cmd, 1000 / portTICK_RATE_MS);
+  i2c_cmd_link_delete(cmd);
+
+  if (ret != ESP_OK)
+  {
+    ESP_LOGW(TAG, "I2C ERROR");
+  }
+
+  vTaskDelay(1 / portTICK_RATE_MS);
+  cmd = i2c_cmd_link_create();
+  i2c_master_start(cmd);
+  i2c_master_write_byte(cmd, TAS5805M_ADDRESS << 1 | READ_BIT, ACK_CHECK_EN);
+  i2c_master_read_byte(cmd, data, NACK_VAL);
+  i2c_master_stop(cmd);
+  ret = i2c_master_cmd_begin(I2C_TAS5805M_MASTER_NUM, cmd, 1000 / portTICK_RATE_MS);
+  i2c_cmd_link_delete(cmd);
+
+  return ret;
+}
+
+// Writing of TAS5805M-Register
+
+esp_err_t tas5805m_write_byte(uint8_t register_name, uint8_t value)
+{
+  int ret;
+  i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+  i2c_master_start(cmd);
+  i2c_master_write_byte(cmd, TAS5805M_ADDRESS << 1 | WRITE_BIT, ACK_CHECK_EN);
+  i2c_master_write_byte(cmd, register_name, ACK_CHECK_EN);
+  i2c_master_write_byte(cmd, value, ACK_CHECK_EN);
+  i2c_master_stop(cmd);
+  ret = i2c_master_cmd_begin(I2C_TAS5805M_MASTER_NUM, cmd, 1000 / portTICK_RATE_MS);
+  i2c_cmd_link_delete(cmd);
+
+  return ret;
+}
+
+// Inits the TAS5805M change Settings in Menuconfig to enable Bridge-Mode
+
+esp_err_t tas5805m_init()
+{
+  int ret;
+  // Init the I2C-Driver
+  i2c_master_init();
+  /* Register the PDN pin as output and write 1 to enable the TAS chip */
+  /* TAS5805M.INIT() */
+  gpio_config_t io_conf;
+  io_conf.intr_type = GPIO_INTR_DISABLE;
+  io_conf.mode = GPIO_MODE_OUTPUT;
+  io_conf.pin_bit_mask = TAS5805M_GPIO_PDN_MASK;
+  io_conf.pull_down_en = GPIO_PULLDOWN_DISABLE;
+  io_conf.pull_up_en = GPIO_PULLUP_DISABLE;
+  ESP_LOGW(TAG, "Power down pin: %d", TAS5805M_GPIO_PDN);
+  gpio_config(&io_conf);
+  gpio_set_level(TAS5805M_GPIO_PDN, 0);
+  vTaskDelay(200 / portTICK_RATE_MS);
+  gpio_set_level(TAS5805M_GPIO_PDN, 1);
+  vTaskDelay(200 / portTICK_RATE_MS);
+
+  /* TAS5805M.Begin()*/
+
+  ESP_LOGW(TAG, "Setting to HI Z");
+  ret = tas5805m_write_byte(TAS5805M_DEVICE_CTRL_2_REGISTER, 0x02);
+  vTaskDelay(100 / portTICK_RATE_MS);
+  if (ret != ESP_OK)
+    return ret;
+  ESP_LOGW(TAG, "Setting to PLAY");
+  ret = tas5805m_write_byte(TAS5805M_DEVICE_CTRL_2_REGISTER, 0x03);
+  if (ret != ESP_OK)
+    return ret;
+
+    // Check if Bridge-Mode is enabled
+#ifdef CONFIG_DAC_BRIDGE_MODE
+  uint8_t value = 0;
+  ret = tas5805m_read_byte(TAS5805M_DEVICE_CTRL_1_REGISTER, &value);
+  if (ret != ESP_OK)
+    return ret;
+  value = 0b100;
+  // ESP_LOGW(TAG,"Setting to MONO mode, CTRL_1 = %d", value);
+  ret = tas5805m_write_byte(TAS5805M_DEVICE_CTRL_1_REGISTER, value);
+  if (ret != ESP_OK)
+    return ret;
+#endif
+
+  vTaskDelay(100 / portTICK_RATE_MS);
```

Comment:
Needed?

Reply:
@CarlosDerSeher .. I will reduce them to 50micro seconds. Hope it works then anyways.